p4-16/spec/P4-16-spec.mdk

```diff
@@ -4465,7 +4472,15 @@ parser Tcp_option_parser(packet_in b, out Tcp_option_stack vec) {
 }
 ~End P4Example
 
-Two header unions can be compared for equality (==) or inequality (!=)
+An expression of the form '(' `type` ')' '{#}', where `type` is a header union type,
+denotes an invalid header union with the specified type.
+
+~ Begin P4Example
+header_union HU { ... }
+HU h = (HU){#};  // invalid header union; same as an uninitialized header union.
```

Comment:
This is a fairly nit-pick comment, but this assignment is only the "same as an uninitialized header union" if that header union is defined as an `out` direction parameter of the enclosing parser or control, but it is only the same in that case BECAUSE the spec defines all such `out` headers and header_unions to be initialized to invalid.

If you declared a local variable of type header or header_union, I believe the spec defines NOTHING about the initial value.  That is, for such a local varriable, even h.isValid() would return an unspecified true or false value, whatever an implementation decides to do.

Reply:
I would thus recommend simply removing "same as an uninitialized header union" from the comment, leaving it to be only "invalid header union" and nothing else.

Reply:
Hmmm.   On a re-read of the spec regarding header_union, I now notice this statement at the beginning of the section "Operations on header unions" that I do not recall noticing before, but it has been there since 2017:

```
A variable declared with a union type is initially invalid.
```
That appears to be _different_ than the initialization behavior for normal headers, but I have no suggestion of changing it now.

Please disregard my earlier comments that are at odds with the language spec earlier on this point.